db/migrate/20220201160946_create_contacts.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateContacts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :contacts do |t|
+      t.string :name
+      t.date :bith_date
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+      add_index :contacts, %i[name user_id], unique: true
```

Comment:
Você pode usar o `t` dentro do bloco para adicionar o índice

```suggestion
      t.index %i[name user_id], unique: true
```